### PR TITLE
(docs): Added for pod network corruption experiment

### LINKED ISSUE #1059: (Docs): Add Docs for pod network packet corruption experiment
**What needs to be done**

-  Add docs for pod network packet corruption experiment in litmuschoas/litmus-docs repository.

### PR DESCRIPTION
**Pull request by ispeakc0de**
Signed-off-by: shubhamchaudhary <shubham.chaudhary@mayadata.io>

- Fixes: litmuschaos/litmus#1059

### CODE REVIEW

File: docs/pod-network-corruption.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+## Experiment Metadata
+
+| Type      | Description              | Tested K8s Platform                                               |
+| ----------| ------------------------ | ------------------------------------------------------------------|
+| Generic   | Inject Network Packet Corruption Into Application Pod | GKE, Konvoy(AWS), Packet(Kubeadm), OpenShift(Baremetal) , Minikube > v1.6.0 |
+
+## Prerequisites
+- Ensure that the Litmus Chaos Operator is running. If not, install from [here](https://github.com/litmuschaos/chaos-operator/blob/master/deploy/operator.yaml)
+- Ensure that the `pod-network-corruption` experiment resource is available in the cluster. If not, install from [here](https://hub.litmuschaos.io/charts/generic/experiments/pod-network-corruption)
+
```
Comment (ksatchit) [inline docs/pod-network-corruption.md]: command sample to check presence of chaosexperiment

File: docs/pod-network-corruption.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+- Application pods are healthy before chaos injection
+
+## Exit Criteria
+
+- Application pods are healthy post chaos injection
+
+## Details
+
+- The application pod should be healthy once chaos is stopped. Service-requests should be         served despite chaos.
+- Injects packket corruption on the specified container by starting a traffic control (tc) process with netem rules to add egress packet corruption
```
Comment (ksatchit) [inline docs/pod-network-corruption.md]:
```suggestion
- Injects network packet corruption on the specified container by starting a traffic control (tc) process with appropriate netem rules
```

File: docs/pod-network-corruption.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+- Corruption is injected via pumba library with command pumba netem corruption by passing the relevant network interface, packet-corruption-percentage, chaos duration and regex filter for container name
```
Comment (ksatchit) [inline docs/pod-network-corruption.md:35]:
```suggestion
- Corruption is injected via pumba library with command `pumba netem corruption` by passing the relevant network interface, packet-corruption-percentage, chaos duration and regex filter for container name
```

File: docs/pod-network-corruption.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+- Override the experiment tunables if desired
+
+#### Supported Experiment Tunables
+
+| Variables             | Description                                                  | Type      | Notes                                                      |
+| ----------------------| ------------------------------------------------------------ |-----------|------------------------------------------------------------|
+| NETWORK_INTERFACE     | Name of ethernet interface considered for shaping traffic                                | Mandatory  |   |
+| TARGET_CONTAINER     | Name of container which is subjected to network latency       | Mandatory  |   |
+| TOTAL_CHAOS_DURATION  | The time duration for chaos insertion in milliseconds  | Optional| Default (60000ms)|
+| NETWORK_PACKET_CORRUPTION_PERCENTAGE       | Packet corruption in percentage                           | Mendatory | Default (100)
```
Comment (ksatchit) [inline docs/pod-network-corruption.md]:
```suggestion
| NETWORK_PACKET_CORRUPTION_PERCENTAGE       | Packet corruption (in percentage)                          | Mandatory | Default (100)
```

File: docs/pod-network-corruption.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+#### Sample ChaosEngine Manifest
+
+```yaml
+apiVersion: litmuschaos.io/v1alpha1
+kind: ChaosEngine
+metadata:
+  name:  nginx-network-chaos
+  namespace: default
+spec:
+  jobCleanUpPolicy: retain
```
Comment (ksatchit) [inline docs/pod-network-corruption.md:71]:
```suggestion
  jobCleanUpPolicy: delete
```

File: docs/pod-network-corruption.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+#### Supported Experiment Tunables
+
+| Variables             | Description                                                  | Type      | Notes                                                      |
+| ----------------------| ------------------------------------------------------------ |-----------|------------------------------------------------------------|
+| NETWORK_INTERFACE     | Name of ethernet interface considered for shaping traffic                                | Mandatory  |   |
+| TARGET_CONTAINER     | Name of container which is subjected to network latency       | Mandatory  |   |
+| TOTAL_CHAOS_DURATION  | The time duration for chaos insertion in milliseconds  | Optional| Default (60000ms)|
+| NETWORK_PACKET_CORRUPTION_PERCENTAGE       | Packet corruption in percentage                           | Mendatory | Default (100)
+| LIB                   | The chaos lib used to inject the chaos eg. Pumba             | Optional  |  |
+| CHAOSENGINE     | ChaosEngine CR name associated with the experiment instance      | Optional  |   |
```
Comment (ksatchit) [inline docs/pod-network-corruption.md:57]: LIB_IMAGE

File: docs/pod-network-corruption.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+```
+### Create the ChaosEngine Resource
+
+- Create the ChaosEngine manifest prepared in the previous step to trigger the Chaos.
+
+  `kubectl apply -f chaosengine.yml`
+
+### Watch Chaos progress
+
+- View network latency by setting up a ping on the affected pod from the cluster nodes
```
Comment (ksatchit) [inline docs/pod-network-corruption.md]: View impact of network packet corruption on the affected pod from the cluster nodes (alternate is to setup ping to a remote IP from inside the target pod)

File: docs/pod-network-corruption.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+title: Pod Network Corruption Experiment Details
+sidebar_label: Pod Network Corruption
+---
+------
+
+## Experiment Metadata
+
+| Type      | Description              | Tested K8s Platform                                               |
+| ----------| ------------------------ | ------------------------------------------------------------------|
+| Generic   | Inject Network Packet Corruption Into Application Pod | GKE, Konvoy(AWS), Packet(Kubeadm), OpenShift(Baremetal) , Minikube > v1.6.0 |
```
Comment (ksatchit) [inline docs/pod-network-corruption.md]: remove openshift

File: docs/pod-network-corruption.md [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+- Application pods are healthy before chaos injection
+
+## Exit Criteria
+
+- Application pods are healthy post chaos injection
+
+## Details
+
+- The application pod should be healthy once chaos is stopped. Service-requests should be         served despite chaos.
+- Injects packket corruption on the specified container by starting a traffic control (tc) process with netem rules to add egress packet corruption
```
Comment (ksatchit) [inline docs/pod-network-corruption.md]: packet

**Review by ksatchit — APPROVED** [review]

**Review by chandankumar4 — APPROVED** [review]

**Pull request merged**